include/persistent_data.hpp

```diff
@@ -188,6 +189,84 @@ class ConfigFile
         }
     }
 
+#ifdef BMCWEB_ENABLE_IBM_MANAGEMENT_CONSOLE
+    void writeCurrentSessionData()
+    {
+        std::ofstream persistentFile(dumpFilename);
+        std::filesystem::perms permission =
+            std::filesystem::perms::owner_read |
+            std::filesystem::perms::owner_write |
+            std::filesystem::perms::group_read;
+        std::filesystem::permissions(filename, permission);
+        const auto& eventServiceConfig =
+            EventServiceStore::getInstance().getEventServiceConfig();
+        nlohmann::json data{
+            {"eventservice_config",
+             {{"ServiceEnabled", eventServiceConfig.enabled},
+              {"DeliveryRetryAttempts", eventServiceConfig.retryAttempts},
+              {"DeliveryRetryIntervalSeconds",
+               eventServiceConfig.retryTimeoutInterval}}
+
+            },
+            {"system_uuid", systemUuid},
+            {"revision", jsonRevision},
+            {"timeout", SessionStore::getInstance().getTimeoutInSeconds()}};
+
+        nlohmann::json& sessions = data["sessions"];
+        sessions = nlohmann::json::array();
+        for (const auto& p : SessionStore::getInstance().authTokens)
+        {
+            if (p.second->persistence !=
+                persistent_data::PersistenceType::SINGLE_REQUEST)
+            {
+                sessions.push_back({
+                    {"unique_id", p.second->uniqueId},
+                    {"username", p.second->username},
+                    {"client_ip", p.second->clientIp},
+                    {"client_id", p.second->clientId},
+                });
```

Comment:
This would be better to be done like https://github.com/ibm-openbmc/bmcweb/blob/292e2f278ecdcd62c14b744b19c0b6451eefe95d/include/persistent_data.hpp#L228

```
               nlohmann::json::object_t session;
                session["unique_id"] = p.second->uniqueId;
               ...
                sessions.emplace_back(std::move(session));
```

I think the current code may cause the compiler error....but not sure why it did not hit - perhaps it is because `ibm-management-console` is not defined.

Reply:
+1